### ISSUE TITLE
Wrong `AttemptResult` type used in `after:run` hook

### ISSUE THREAD

**Issue by csvtuda** [cypress-io/cypress#27390]
### Current behavior

Following the release of version 12.17.0, the types of certain plugin event hooks have been modified as mentioned in https://github.com/cypress-io/cypress/issues/24110. The corresponding PR successfully fixed the types of the `after:spec` hook.

However, this also affected the `after:run` hook, which ***also*** uses the `AttemptResult` type. The main issue is that the attempts of the `after:run` hook do ***not*** contain properties `wallClockStartedAt` or `wallClockDuration`, but still use `startedAt` and `duration`. 

I have attached test code to reproduce the problem as well as its output. It is clear to see that the `after:run` hook still provides objects with `startedAt` and `duration`.

Unfortunately, because type `AttemptResult` is used by both hooks and now only contains properties `wallClockStartedAt` and `wallClockDuration`, code which tries to access the `startedAt` and `duration` properties inside the `after:run` hook cannot be compiled in (properly typed) typescript currently.

### Desired behavior

It would be great to have Cypress either:
- provide two different `AttemptResult` types, one for each hook
- replace `duration` and `startedAt` inside the `after:run` attempt objects with `wallClockDuration` and `wallClockStartedAt`

### Test code to reproduce

```ts
export default (on: Cypress.PluginEvents, config: Cypress.PluginConfigOptions) => {
  on("after:spec", (spec: Cypress.Spec, results: CypressCommandLine.RunResult) => {
    const test = results.tests[0];
    const attempt = test.attempts[0];
    console.log("after:spec");
    console.log(attempt);
    console.log(attempt.wallClockDuration);
    console.log(attempt.wallClockStartedAt);
  });
  on("after:run", (results: CypressCommandLine.CypressRunResult | CypressCommandLine.CypressFailedRunResult) => {
    const run = (results as CypressCommandLine.CypressRunResult).runs[0];
    const test = run.tests[0];
    const attempt = test.attempts[0];
    console.log("after:run");
    console.log(attempt);
    console.log(attempt.duration);  // Error: Property 'duration' does not exist on type 'AttemptResult'
    console.log(attempt.startedAt); // Error: Property 'startedAt' does not exist on type 'AttemptResult'
  });
}
```

### Cypress Version

12.17.0

### Node version

v18.16.0

### Operating System

Windows 1019045.3208

### Debug Logs

```shell
after:spec
{
  state: 'passed',
  error: null,
  timings: { lifecycle: 25, test: { fnDuration: 535, afterFnDuration: 0 } },
  failedFromHookId: null,
  wallClockStartedAt: '2023-07-25T23:24:22.540Z',
  wallClockDuration: 571,
  videoTimestamp: null
}
571
2023-07-25T23:24:22.540Z


after:run
{
  state: 'passed',
  error: null,
  videoTimestamp: null,
  duration: 152,
  startedAt: '2023-07-25T23:24:16.470Z',
  screenshots: []
}
152
2023-07-25T23:24:16.470Z
```

**Comment by ramosbugs** [comment]
I can confirm this issue. The relevant code is: https://github.com/cypress-io/cypress/blob/fe54cf504aefcfa6b621a90baa57e345cfa09548/packages/server/lib/modes/run.ts#L882-L884

**Comment by ramosbugs** [comment]
It looks like this issue got significantly worse in Cypress 13 as a result of #27230/#23805 (cc: @chrisbreiding). With the latest release, `state` is now the *only* field reported for each test attempt: https://github.com/cypress-io/cypress/blob/0f9a150647146c3e63e14d298d7fcf7ad5c92b66/packages/server/lib/modes/results.ts#L111

I think there was an assumption made without gathering user feedback that per-attempt timestamps aren't used by Cypress users and third-party plugins, but they very much are! Please add these fields back.

**Comment by jennifer-shehane** [comment]
Hi @ramosbugs, we've been trying to clean up the Public APIs and one of the actions we took was keeping the state property only for each attempt. Could you please open a new issue describing your use case for the attempts startedAt/duration fields and a request to add those fields back? We are open to considering adding fields back. 

I'll close this issue itself since it's no longer relevant to the state of v13.
